### PR TITLE
Aggregator by Default - 2.0

### PR DESCRIPTION
**Pull request by michaelmdresser**
## Does this PR relate to any other PRs?
- https://github.com/kubecost/kubecost-cost-model/pull/2001
- https://github.com/kubecost/kubecost-cost-model/pull/200
- Requires https://github.com/kubecost/kubecost-cost-model/pull/2002
- Pairs with https://github.com/opencost/opencost/pull/2429
- Pairs with https://github.com/kubecost/kubecost-cost-model/pull/2020
- Replaces https://github.com/kubecost/cost-analyzer-helm-chart/pull/2886, see for original PR review

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
- Kubecost's new query backend "Aggregator" now serves API requests in all configurations. You will notice a new container running in the cost-analyzer Pod by default. If you are an Enterprise Thanos user, follow [this guide TODO]() for necessary configuration changes. If you are an Enterprise Federated ETL user, follow [this guide TODO]() for necessary configuration changes. If you are a Free user, no configuration change is required. This does _not_ introduce a new container image.
- Cloud Cost data gathering now runs in a separate container (within cost-analyzer) by default. If using an Enterprise Aggregator configuration, Cloud Cost will run in a one-replica Deployment. No configuration or value change is required. This does _not_ introduce a new container image.
- Removes the "HA" (and associated StatefulSet) mode for cost-analyzer. To deploy a highly-available backend, contact Kubecost Support about using the Enterprise Aggregator configuration with replicas > 1. If the cost-model's low-footprint metric building has an HA requirement in your environment, contact Kubecost Enterprise Support so we can learn more.
- The `federatedETL.primaryCluster` value has been removed. In an Enterprise Federated ETL configuration all cost-model containers should behave as if they were secondaries. Aggregator now serves combined queries. If you are an Enterprise Federated ETL user, follow [this guide TODO]() for necessary configuration changes.
- The Federator component (enabled via `federatedETL.federator.enabled`) has been removed. In an Enterprise Federated ETL configuration all queries and data-combination is handled by Aggregator. If you are an Enterprise Federated ETL user, follow [this guide TODO]() for necessary configuration changes.
- The `kubecostModel.warmSavingsCache` value has been removed. If you need to disable the savings cache in Aggregator, set the env var `SAVINGS_ENABLED` to `false`.
- Query Service (Replicas) has been removed. Aggregator now takes its place, serving queries faster with a lower resource footprint. If you are an Enterprise Federated ETL user, follow [this guide TODO]() for necessary configuration changes.
- Unsupported Helm values for an old metric storage method have been removed. No impact is expected.

## Links to Issues or tickets this PR addresses or fixes
- Main PR for https://kubecost.atlassian.net/browse/SELFHOST-1043

## Have you made an update to documentation? If so, please provide the corresponding PR.
WIP

### CODE REVIEW

**Comment by michaelmdresser** [comment]
This PR also fixes some small problems with the current state of the 2.0 branch.

**Comment by michaelmdresser** [comment]
@chipzoller Rebased onto 2.0 and opened this PR. CI cannot pass because of the 1.108 images it is using.

**Comment by chipzoller** [comment]
@michaelmdresser, we just had a meeting where this came up. @AjayTripathy is recommending that you run the equivalent steps in the CI process by pointing the chart at the nightly latest images to ensure this comes up. Check the [CI manifest](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/.github/workflows/chart.yaml) for what checks are running (`helm test` can help make this easier on you).

File: cost-analyzer/templates/NOTES.txt [hunk 1]
```diff
@@ -1,7 +1,10 @@
 
 
 --------------------------------------------------
-{{- if and (.Values.thanos) (.Values.thanos.store) (.Values.thanos.store.enabled) }}
+{{/*
+https://github.com/helm/helm/issues/8026#issuecomment-881216078
+*/}}
+{{- if ((.Values.thanos).store).enabled }}
```
Comment (chipzoller) [inline cost-analyzer/templates/NOTES.txt:7]: Thank you, Michael. This is indeed much nicer.

**Review by chipzoller — APPROVED** [review]
Approved with the assumption that manual testing, analogous to those which run in CI here, is successful against nightly images.

**Comment by michaelmdresser** [comment]
Thank you again for the super thorough review, let's talk if anything unexpected happens after the merge.

**Pull request merged**